### PR TITLE
Solution to issue 536 where record values are sometimes placed in offside

### PR DESCRIPTION
**Pull request by drhumlen**

### CODE REVIEW

File: src/Fantomas/CodePrinter.fs [hunk 1]
```diff
@@ -867,14 +867,24 @@ and genExpr astContext synExpr =
     | Record(inheritOpt, xs, eo) ->
         let recordExpr =
             let fieldsExpr = col sepSemiNln xs (genRecordFieldName astContext)
+            let fieldsExprShortForm = (!- " with " +> fieldsExpr)
```
Comment (drhumlen) [inline src/Fantomas/CodePrinter.fs:870]: I need to do this for anon records too

File: src/Fantomas/CodePrinter.fs [hunk 2]
```diff
@@ -867,14 +867,24 @@ and genExpr astContext synExpr =
...
             eo |> Option.map (fun e ->
-                genExpr astContext e +> ifElseCtx (futureNlnCheck fieldsExpr) (!- " with" +> indent +> sepNln +> fieldsExpr +> unindent) (!- " with " +> fieldsExpr))
+                genExpr astContext e +> ifElseCtx (futureNlnCheck fieldsExprShortForm)
+                                                  (!- " with" +> indent +> sepNln +> fieldsExpr +> unindent)
+                                                  fieldsExprShortForm)
             |> Option.defaultValue fieldsExpr
 
+        let someSeparator =
```
Comment (drhumlen) [inline src/Fantomas/CodePrinter.fs:877]: This is a bad name, but I think breaking up the code below a little is necessary.

File: src/Fantomas/Context.fs [hunk 3]
```diff
@@ -62,46 +62,46 @@ module WriterModel =
...
       WriterEvents : Queue<WriterEvent>;
       BreakLines : bool;
       BreakOn : string -> bool;
-      /// The original source string to query as a last resort 
-      Content : string; 
+      /// The original source string to query as a last resort
+      Content : string;
       /// Positions of new lines in the original source string
-      Positions : int []; 
+      Positions : int [];
```
Comment (drhumlen) [inline src/Fantomas/Context.fs:87]: Sorry for touching so many lines, but the file was riddled with trailing whitespaces. It has to go.

Hopefully, if we start formatting the codebase itself with fantomas, there won't be any unrelated formatting fixes in the future! 😄

**Comment by nojaf** [comment]
Hey @drhumlen, 

Thanks for this PR.
I'm always happy that people want to solve their own problem so cheers for that!

Your changes are colliding with the work I'm doing in #721 I'm afraid.
See https://github.com/fsprojects/fantomas/pull/721/files#diff-a1b0b9a30fd94fe61b35a1a52ce27955R857

In that branch, the user can decide how long a short record can be.
So I already touched that part of `CodePrinter` quite heavily.

I think I'm getting closer to finishing #721 and I'd like to cherry-pick from your PR once that is merged. I hope you understand.

Cheers,

nojaf

**Comment by drhumlen** [comment]
@nojaf Ah, yea that's fine 😄. I can rebase my branch against master once it's merged.

Do you have any ETA for when you'll be merging your branch https://github.com/fsprojects/fantomas/pull/721?

**Comment by nojaf** [comment]
ETA is hard to say, I think maybe somewhere near the end of this week.
I'm actively working on it though.

**Comment by drhumlen** [comment]
@nojaf : I see you've merged the PR. I will merge in the changes to my branch

**Comment by drhumlen** [comment]
I see the unit tests I've added on broken formatting are now passing after accepting the changes from master on merge conflicts. So it seems some of the work I've done is now already solved in the new code in master.

@nojaf: The unit tests are still nice to have though unless you've already tested the same behaviour/fix elsewhere? If not, this PR contains mostly a few renames and removal of trailing whitespace. I'm not sure if it's worth merging or not..?

**Pull request closed without merging**